semgrep/rxjs-subscription.yaml

```diff
@@ -116,3 +116,11 @@ rules:
     message: ngOnDestroy missing this.isDestroyed.complete()
     languages: [ts]
     severity: ERROR
+
+  - id: onDestroy-not-implemented-in-class
+    patterns:
+    - pattern: this.isDestroyed.complete()
+    - pattern-not-inside: class $CLASS implements OnDestroy { ... }
+    message: subscriptions never complete because class does not implement OnDestroy Lifecycle hook
```

Comment:
That is, even if there is an `ngOnDestroy()` defined, it is never used without the `implements`.